core/src/main/java/io/questdb/griffin/engine/functions/date/VarcharToDateFunctionFactory.java

```diff
@@ -0,0 +1,99 @@
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.date;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.sql.Function;
+import io.questdb.cairo.sql.Record;
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.PlanSink;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.DateFunction;
+import io.questdb.griffin.engine.functions.UnaryFunction;
+import io.questdb.std.IntList;
+import io.questdb.std.Numbers;
+import io.questdb.std.NumericException;
+import io.questdb.std.ObjList;
+import io.questdb.std.datetime.DateFormat;
+import io.questdb.std.datetime.DateLocale;
+import io.questdb.std.datetime.millitime.DateFormatCompiler;
+import io.questdb.std.str.Utf8Sequence;
+
+public class VarcharToDateFunctionFactory implements FunctionFactory {
+    private static final ThreadLocal<DateFormatCompiler> tlCompiler = ThreadLocal.withInitial(DateFormatCompiler::new);
+
+    @Override
+    public String getSignature() {
+        return "to_date(Øs)";
+    }
+
+    @Override
+    public Function newInstance(int position, ObjList<Function> args, IntList argPositions, CairoConfiguration configuration, SqlExecutionContext sqlExecutionContext) throws SqlException {
+        final Function arg = args.getQuick(0);
+        final CharSequence pattern = args.getQuick(1).getStrA(null);
+        if (pattern == null) {
+            throw SqlException.$(argPositions.getQuick(1), "pattern is required");
+        }
+        return new ToDateFunction(arg, tlCompiler.get().compile(pattern), configuration.getDefaultDateLocale(), pattern);
+    }
+
+    private static final class ToDateFunction extends DateFunction implements UnaryFunction {
+
+        private final Function arg;
+        private final DateFormat dateFormat;
+        private final DateLocale locale;
+        private final CharSequence pattern;
+
+        public ToDateFunction(Function arg, DateFormat dateFormat, DateLocale locale, CharSequence pattern) {
+            this.arg = arg;
+            this.dateFormat = dateFormat;
+            this.locale = locale;
+            this.pattern = pattern;
+        }
+
+        @Override
+        public Function getArg() {
+            return arg;
+        }
+
+        @Override
+        public long getDate(Record rec) {
+            Utf8Sequence value = arg.getVarcharA(rec);
+            try {
+                if (value != null && value.isAscii()) {
+                    return dateFormat.parse(value.asAsciiCharSequence(), locale);
```

Comment:
This function won't work correctly with non-English (non-ASCII) `DateLocale`. We need to decode UTF-16 to UTF-8 when the sequence is not ASCII. It would be great to add a test around that.

Reply:
Thanks, I changed to handle non-en locale case